lightning/src/ln/channel.rs

```diff
@@ -434,6 +434,15 @@ pub(super) struct Channel<Signer: Sign> {
 	next_local_commitment_tx_fee_info_cached: Mutex<Option<CommitmentTxInfoCached>>,
 	#[cfg(any(test, feature = "fuzztarget"))]
 	next_remote_commitment_tx_fee_info_cached: Mutex<Option<CommitmentTxInfoCached>>,
+
+	/// lnd has a long-standing bug where, upon reconnection, if the channel is not yet confirmed
+	/// they will not send a channel_reestablish until the channel locks in. Then, they will send a
+	/// funding_locked *before* sending the channel_reestablish (which is clearly a violation of
+	/// the BOLT specs). We copy c-lightning's workaround here and simply store the funding_locked
```

Comment:
```
A node:

    if next_commitment_number is 1 in both the channel_reestablish it sent and received:
        MUST retransmit funding_locked.
    otherwise:
        MUST NOT retransmit funding_locked.
```

I agree, it's pretty clear, though another fix for them would be to rebroadcast a redundant `funding_locked` after `channel_reestablish`.

Reply:
Hmm, no, if they don't know what is in `channel_reestablish` they "MUST NOT retransmit funding_locked." I don't think they could do that either.